server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1721,6 +1770,10 @@ public ShardGenerations getUpdatedShardGenerations() {
         }
 
         public Iterator<String> getBlobPaths() {
+            if (compressed == null || shardDeleteResults == null) {
+                // No output stream: nothing to return
```

Comment:
Likewise here I don't think we should change anything with respect to these values being `null`.